src/usbg_common.c

```diff
@@ -23,6 +23,77 @@
 #include <unistd.h>
 #include <sys/sysmacros.h>
 
+int usbg_write_guid(const char *path, const char *name,
+		      const char *file, const char *buf)
+{
+	uint8_t guidbin[GUID_BIN_LENGTH];
+	int ret;
+
+	if (strlen(buf) != GUID_CHAR_LENGTH)
+		return USBG_ERROR_INVALID_PARAM;
+
+	/* scan actual characters */
+	/* XXXXXXXX-YYYY-YYYY-YYYY-YYYYYYYYYYYY */
+	for (int i = 0; i < 4; i++) {
+		sscanf(buf, "%02x", &guidbin[i]);
+		buf += 2;
+	}
+
+	/* check for next char to be '-' */
+	if (*buf != '-')
+		return USBG_ERROR_INVALID_PARAM;
+	buf += 1;
+
+	/* scan actual characters */
+	/* YYYYYYYY-XXXX-YYYY-YYYY-YYYYYYYYYYYY */
+	for (int i = 0; i < 2; i++) {
+		sscanf(buf, "%02x", &guidbin[4 + i]);
+		buf += 2;
+	}
+
+	/* check for next char to be '-' */
+	if (*buf != '-')
+		return USBG_ERROR_INVALID_PARAM;
+	buf += 1;
+
+	/* scan actual characters */
+	/* YYYYYYYY-YYYY-XXXX-YYYY-YYYYYYYYYYYY */
+	for (int i = 0; i < 2; i++) {
+		sscanf(buf, "%02x", &guidbin[6 + i]);
+		buf += 2;
+	}
+
+	/* check for next char to be '-' */
+	if (*buf != '-')
+		return USBG_ERROR_INVALID_PARAM;
+	buf += 1;
+
+	/* scan actual characters */
+	/* YYYYYYYY-YYYY-YYYY-XXXX-YYYYYYYYYYYY */
+	for (int i = 0; i < 2; i++) {
+		sscanf(buf, "%02x", &guidbin[8 + i]);
+		buf += 2;
+	}
+
+	/* check for next char to be '-' */
+	if (*buf != '-')
+		return USBG_ERROR_INVALID_PARAM;
+	buf += 1;
+
+	/* scan actual characters */
+	/* YYYYYYYY-YYYY-YYYY-YYYY-XXXXXXXXXXXX */
+	for (int i = 0; i < 6; i++) {
+		sscanf(buf, "%02x", &guidbin[10 + i]);
+		buf += 2;
+	}
```

Comment:
```suggestion	
	ret = sscanf(buf, "%02hhx%02hhx%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx%02hhx%02hhx%02hhx%02hhx",
 	        &guidbin[0], &guidbin[1], &guidbin[2], &guidbin[3],
  	        &guidbin[4], &guidbin[5], &guidbin[6], &guidbin[7],
	        &guidbin[8], &guidbin[9], &guidbin[10], &guidbin[11],
	        &guidbin[12], &guidbin[13], &guidbin[14], &guidbin[15]);
	if (ret != GUID_BIN_LENGTH)
        	return USBG_ERROR_INVALID_PARAM;
```

Reply:
Uugh, but sure, snprintf and sscanf is just ugly. Thanks for the changes! I will add them.

Reply:
Btw. Would %02x not be enough? I don't think the %02hhx is necessary. This would significantly resize the parsing string.